ruby/engine/test/RubyEngine_GTest.cpp

```diff
@@ -0,0 +1,106 @@
+/***********************************************************************************************************************
+*  OpenStudio(R), Copyright (c) Alliance for Sustainable Energy, LLC.
+*  See also https://openstudio.net/license
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+
+#include "measure/ModelMeasure.hpp"
+#include "measure/OSArgument.hpp"
+#include "measure/OSMeasure.hpp"
+#include "model/Model.hpp"
+#include "scriptengine/ScriptEngine.hpp"
+
+#include <fmt/format.h>
+
+#include <regex>
+
+class RubyEngineFixture : public testing::Test
+{
+ public:
+  static openstudio::path getScriptPath(const std::string& classAndDirName) {
+    openstudio::path scriptPath = openstudio::getApplicationSourceDirectory() / fmt::format("ruby/engine/test/{}/measure.rb", classAndDirName);
+    OS_ASSERT(openstudio::filesystem::is_regular_file(scriptPath));
+    return scriptPath;
+  }
+
+  static std::string stripAddressFromErrorMessage(const std::string& error_message) {
+    static std::regex object_address_re("0x[[:alnum:]]*>");
+
+    return std::regex_replace(error_message, object_address_re, "ADDRESS>");
+  }
+
+ protected:
+  // initialize for each test
+  virtual void SetUp() override {
+    std::vector<std::string> args;
+    thisEngine = std::make_unique<openstudio::ScriptEngineInstance>("rubyengine", args);
+
+    (*thisEngine)->registerType<openstudio::measure::OSMeasure*>("openstudio::measure::OSMeasure *");
+    (*thisEngine)->registerType<openstudio::measure::ModelMeasure*>("openstudio::measure::ModelMeasure *");
+    (*thisEngine)->exec("OpenStudio::init_rest_of_openstudio()");
+  }
+  // tear down after each test
+  virtual void TearDown() override {
+    // Want to ensure the engine is reset regardless of the test outcome, or it may throw a segfault
+    thisEngine->reset();
+  }
+
+  std::unique_ptr<openstudio::ScriptEngineInstance> thisEngine;
```

Comment:
FYI @kbenne I slightly changed the test you wrote so that the fixture will always reset the engine, regardless of whether the test passed or failed.